test/test_compile.py

```diff
@@ -0,0 +1,57 @@
+import argparse
+import os.path
+import pytest
+
+from otk.command import compile, parser_create
+
+
+@pytest.mark.parametrize(
+    "arguments, input_data, output_data, sys_exit_code, log_message",
+    [
+        # "GENERATE" is a placeholder to append "tmp_path" from pytest
+        (
+            argparse.Namespace(input="GENERATE", output="GENERATE", target=None),
+            """otk.version: 1
+otk.target.osbuild.qcow2: { test: 1 }
+""",
+            """{
+  "test": 1,
+  "version": "2",
+  "sources": {}
+}""",
+            0,
+            None,
+        ),
+    ],
+)
+def test_compile(
+    tmp_path,
+    caplog,
+    capsys,
+    monkeypatch,
+    arguments,
+    input_data,
+    output_data,
+    sys_exit_code,
+    log_message,
+):
+    if arguments.input == "GENERATE":
```

Comment:
(super nitpick) I wonder if the test parameter really needs to be a full argparse.Namespace, it seems we are only intrested in "file" or "stdin/stdout", maybe a combinartion. So just passing "input", "output" as either "some-filename" or "-" (for stdin/stdout) might be slightly shorter and then here we would construct the namespace